### PR TITLE
chore: add typescript

### PR DESCRIPTION
**Pull request by vietnogigoodrx**
### 📃 Summary
Add minimal settings to allow typescript to run.  Will convert each files and add better typings as time goes on.

### 🧪 QA
Pull down code and run `yarn dev` and everything should compile.  Modify icon.tsx and changes should show up in the browser.  If an error occurs, it should show an error in the browser.  `yarn build` and `yarn test` should continue working.
	
### 📈 Risk Analysis
- [ ] Does this affect patients?
- [ ] Does this affect clinical?
- [ ] Does this mutate data?
- [ ] Does it touch PHI?
- [ ] Does it touch payments?
- [ ] Does it update Node packages?

### CODE REVIEW

File: .storybook/presets.js [hunk 1]
```diff
@@ -1 +0,0 @@
-module.exports = ['@storybook/addon-docs/react/preset'];
```
Comment (vietnogigoodrx) [inline .storybook/presets.js:1]: added to main.js

File: package.json [hunk 2]
```diff
@@ -16,8 +16,9 @@
   "scripts": {
     "clean": "rimraf lib",
     "watch": "yarn rollup --watch",
-    "build": "npm run clean && babel src --out-dir lib",
+    "build": "npm run clean && babel src --out-dir lib --extensions '.ts,.tsx,.js,.jsx'",
```
Comment (schlegz) [inline package.json:19]: Curious why we include the `jsx` extension here but not in the extensions list in the other config files.

Comment (vietnogigoodrx) [reply package.json:19]: missed it in the other file, added

**Comment by vietnogigoodrx** [comment]
Can I get somebody to review this PR?

**Comment by vietnogigoodrx** [comment]
> Remove `package-lock.json` since we moved to using `yarn.lock`.

did it in another PR

**Comment by albertovillalobos** [comment]
@vietnogigoodrx answering your quesiton here as well:

The check that says “complete setup” you can ignore because this branch was cut before chromatic changes made it in
The other one looks like a component actually changed
and needs to be approved, it looks like you “created” a new icon (actually just changed extension), So it’s detected as a new change, which i’ll approve

**Pull request merged**